### PR TITLE
fix(dropdown): added ngbdropdown to have dropdown class always

### LINKED ISSUE #361: Dropdown: dropdown-menu is not properly aligned with dropdown-toggle button
screen shot 2016-07-05 at 8 33 19 pm

### PR DESCRIPTION
**Pull request by sendilkumarn**
Before submitting a pull request, please make sure you have at least performed the following:

 - [X] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/core/blob/master/CONTRIBUTING.md) guide.
 - [X] built and tested the changes locally.
 - [X] added/updated any applicable tests.
 - [X] added/updated any applicable API documentation.
 - [X] added/updated any applicable demos.

fixes #361

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
This is a good fix but we need to add a check for this class in one of the unit tests.

**Comment by wesleycho** [comment]
Do we want tests for all these classes on host? Wouldn't this theoretically mean we need to do this for the other components too?

**Comment by pkozlowski-opensource** [comment]
> Do we want tests for all these classes on host? Wouldn't this theoretically mean we need to do this for the other components too?

We do have tests for other cases already!

**Comment by wesleycho** [comment]
Ah, was not aware - fair enough then

**Comment by sendilkumarn** [comment]
@pkozlowski-opensource I checked, since there was no check for dropdowntoggle class. I kinda omitted it. Again I think it should be there. I will update

**Comment by sendilkumarn** [comment]
@pkozlowski-opensource ping

**Comment by icfantv** [comment]
@sendilkumarn, please be patient.  we are all busy and Pawel will answer when he has a moment to do so.  he typically bikes home from work around now.

also, you forgot to run the clang formatter so the build failed.

**Comment by sendilkumarn** [comment]
@icfantv just pinged to inform that the issue is fixed. 

point noted :)

File: src/dropdown/dropdown.spec.ts [hunk 1]
```diff
@@ -274,6 +274,28 @@ describe('ngb-dropdown-toggle', () => {
          expect(dropdownEl).toHaveCssClass('open');
        });
      })));
+  it('should have css class dropdown on div and dropdown-toggle on button',
```
Comment (pkozlowski-opensource) [inline src/dropdown/dropdown.spec.ts]: This test should be moved to the ` describe('ngb-dropdown'` section and made much simpler (for example we don't need to have `ngbDropdown` inside at all)

Comment (pkozlowski-opensource) [inline src/dropdown/dropdown.spec.ts]: What I was saying is that we can test the functionality in question with the following markup only: `<div ngbDropdown></div>`

Comment (sendilkumarn) [inline src/dropdown/dropdown.spec.ts]: In that case dropdown-toggle testing functionality will be missing.

Comment (pkozlowski-opensource) [inline src/dropdown/dropdown.spec.ts]: It is covered by other tests already

Comment (sendilkumarn) [inline src/dropdown/dropdown.spec.ts]: Correct me if I am wrong. 

`expect(buttonEl).toHaveCssClass('dropdown-toggle');` 

The `dropdown-toggle` class is also set via host. Ideally there should be a test case right?

This test is not there, am I missing out anything?

Comment (pkozlowski-opensource) [inline src/dropdown/dropdown.spec.ts]: Right, but those are hosts of 2 different directives and we don't need to test them together!

IMO the whole change in this PR should boil down to sth like this: https://github.com/ng-bootstrap/core/commit/7dccb0c37d14c16a793f50d6e8de739952ab05fb

Could you update this PR accordingly?

Comment (sendilkumarn) [inline src/dropdown/dropdown.spec.ts]: Okay.

**Pull request closed without merging**